### PR TITLE
Feat(@tinacms/fields): add labels to toggle field

### PR DESCRIPTION
**Pull request by kendallstrautman**
[Original PR](https://github.com/tinacms/tinacms/pull/1585) by @canrozanes
---
- [x] [docs PR](https://github.com/tinacms/tinacms.org/pull/742) approved
- [x] test case in `demo-next`

### CODE REVIEW

**Comment by kendallstrautman** [comment]
Looks like the field props aren't getting properly passed down, I'll make a PR into this one on a new branch.

File: packages/@tinacms/fields/src/components/Toggle.tsx [hunk 1]
```diff
@@ -29,34 +29,64 @@ export interface ToggleProps {
...
+    false: props.toggleLabels === true ? 'No' : null,
+  }
+
+  if (typeof props.toggleLabels === 'object' && 'true' in props.toggleLabels) {
+    labels.true = props.toggleLabels['true']
+    labels.false = props.toggleLabels['false']
+  }
+
+  const hasToggleLabels =
+    typeof labels.true !== null || typeof labels.false !== null
```
Comment (kendallstrautman) [inline packages/@tinacms/fields/src/components/Toggle.tsx:53]: typeof `null` is always 'object'. We should adjust this to check that the labels equal typeof 'string'

Comment (canrozanes) [reply packages/@tinacms/fields/src/components/Toggle.tsx:53]: good catch. I'll fix it today.

Comment (kendallstrautman) [reply packages/@tinacms/fields/src/components/Toggle.tsx:53]: all good, I can update this as I address some other issues.

Comment (canrozanes) [reply packages/@tinacms/fields/src/components/Toggle.tsx:53]: sure!

**Comment by kendallstrautman** [comment]
[This issue](https://github.com/tinacms/tinacms/issues/1603) affects the toggle field when rendered in the settings modal, but I think it should be addressed in a different PR.

**Comment by kendallstrautman** [comment]
@DirtyF would love a look if you get a chance!

**Comment by kendallstrautman** [comment]
@all-contributors please add @canrozanes for code

**Review by DirtyF — APPROVED** [review]
👌 

You can now set custom labels on `toggle` fields, e.g:

```js
    fields: [
      {
        name: 'Visibility',
        component: 'toggle',
        toggleLabels: {
          true: "show",
          false: "hide"
        }
      },
    ],
```

**Output:**
toggle-custom-labels

Thanks @canrozanes 🙏

**Review by dwalkr — APPROVED** [review]

**Pull request merged**